### DISCUSSION TITLE
middle mouse to pan's behaviour is inconsistent and confusing

### DISCUSSION THREAD

**Discussion by idbrii** [godotengine/godot-proposals#6067] [Editor]
I've used Godot for about two months and I've finally figured out that I need to press Shift to pan the camera.

In a 2D viewport, you middle mouse + drag
and in a 3D viewport, you Shift + middle mouse + drag
but also in a 3D viewport that has View Rotation locked, you can middle mouse + drag
and also sometimes you can right mouse + drag.

I immediately noticed that I could middle mouse click and drag to pan the camera, but then sometimes it wouldn't work. It wasn't obvious why. I was flipping back and forth between a 3D scene and UI, so I guess it worked whenever I was in UI. I think possibly I somehow enabled "Lock View Rotation" because I sometimes panned in the 3D view without Shift.

My most common issue was going to top/left/right view, middle mouse click and drag, and the view rotates. This was frustrating enough to be annoyed, but not enough to disrupt what I'm working on to search for "why does middle click to pan sometimes not work."

I've worked in Unity a lot and [reading their docs](https://docs.unity3d.com/Manual/SceneViewNavigation.html), and they have similar mouse shortcuts except that middle mouse *always pans* (you hold Alt to rotate around something). They also have a more obvious and accessible "Lock View Rotation" button (the 🔒 icon). I haven't worked in 3D modelling software for years, but a lot of 2D art software has middle click to pan.

"Always pan" is easier for new users since it's a simpler mental model to internalize. I can change my editor settings to use None for my pan shortcut and Alt for Orbit. However, that doesn't help new users. Maybe there's another way to make this more consistent.

When you set a top/left/right view, you're trying to work in that 2d view. So what if that auto enabled "Lock View Rotation"? Or if there was an **option that made 2D views auto enable "Lock View Rotation"**?

When you enable "Lock View Rotation", there's a prominent label in the viewport so it shouldn't be too confusing and it allows the behaviour to be more consistent: when the viewport looks 2D, you don't need Shift to pan.

**Comment by Calinou** [comment]
> (you hold Alt to rotate around something). 

This is problematic on Linux, as most window managers use <kbd>Alt</kbd> to perform window manipulation by default. You *cannot* use <kbd>Alt</kbd> + mouse in an application without requiring the user to change their settings, which may be difficult for novice users.

Middle mouse to pan in 3D also matches Blender's behavior. Since a lot of Godot users also use Blender, it makes sense to follow Blender's most commonly used navigation conventions.

> When you set a top/left/right view, you're trying to work in that 2d view. So what if that auto enabled "Lock View Rotation"? Or if there was an option that made 2D views auto enable "Lock View Rotation"?

I think this is already the case when using an orthogonal camera + a rotation preset, but not perspective camera (even after choosing a rotation preset). Again, I think this was done to match Blender.

**Reply by idbrii** [reply]
I've never used Blender, so that's partly why it's unintuitive for me.

> already the case when using an orthogonal camera + a rotation preset

Although an orthogonal camera and a fixed angle camera are quite different views. Do you think of an auto "Lock View Rotation" option? Maybe hard to tell whether it's helpful vs. annoying without actually implementing and using it?
